### PR TITLE
The sso_saml2 code now lives in its own repo.

### PR DESCRIPTION
**Pull request by patrick-hubert-adsk**
Removed the shotgun_shared folder from the authentication package.

Renamed saml2_sso to sso_saml2 and the code now lives in its own repo, with tests and all.

There is a script to update the sso_saml2 package, similar to what is done for the shotgun_api3.

This new version of sso_saml2 now properly supports http_proxies.

### CODE REVIEW

File: developer/sso_saml2_update.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2015 Shotgun Software Inc.
```
Comment (jfboismenu) [inline developer/sso_saml2_update.sh]: 2018

File: python/tank_vendor/sso_saml2/__init__.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2015 Shotgun Software Inc.
```
Comment (jfboismenu) [inline python/tank_vendor/sso_saml2/__init__.py]: Your library has outdated copyrights btw.

Comment (patrick-hubert-adsk) [reply python/tank_vendor/sso_saml2/__init__.py]: Do you know where I could get an updated copyrights header ?  For code and such...

Comment (jfboismenu) [reply python/tank_vendor/sso_saml2/__init__.py]: Just bump the year :)

Comment (patrick-hubert-adsk) [reply python/tank_vendor/sso_saml2/__init__.py]: Pffff.... trop facile... pas de challenge. :-P

**Pull request merged**